logger.go

```diff
@@ -26,6 +26,56 @@ var (
 	disableColor = false
 )
 
+// LoggerConfig defines the config for Logger middleware.
+type LoggerConfig struct {
+	// Optional. Default value is gin.DefaultLogFormatter
+	Formatter LogFormatter
+
+	// Output is a writer where logs are written.
+	// Optional. Default value is gin.DefaultWriter.
+	Output io.Writer
+
+	// SkipPathes is a url path array which logs are not written.
+	// Optional.
+	SkipPathes []string
+}
+
+// LogFormatter gives the signature of the formatter function passed to LoggerWithFormatter
+type LogFormatter func(params LogFormatterParams) string
+
+// LogFormatterParams is the structure any formatter will be handed when time to log comes
+type LogFormatterParams struct {
```

Comment:
I hope what output log which depend on the struct? If I want to add or remove some field, how should I do? thanks!

Reply:
As https://github.com/gin-gonic/gin/issues/1581#issue-366078402 said, want to output some username and payload data.

Reply:
Do you mean what fields should LogFormatterParams provide?

For example, the echo framework's log format also provides user-agent or refer.
(See: https://echo.labstack.com/middleware/logger)

In the this pull's code, You can get them from the Request field, but Gin may be good for adding such fields to LogFormatterParams.

Reply:
If the Gin's user want to add username or payload data, write code into his LogFormatter function which get them by using LogFormatterParams.

Reply:
@sairoutine thank your explain.